src/tests/lvm/lvm_ha_controller/lvm_ha_controller_test.go

```diff
@@ -0,0 +1,274 @@
+
+	// setup sc parameters
+	app.Lvm = k8stest.LvmOptions{
+		VolGroup:      nodeConfig.VgName,
+		Storage:       "lvm",
+		ThinProvision: common.No,
+	}
+
+	// list nodes without NoSchedule taint
+	nodesWithoutTaint, err = k8stest.ListNodesWithoutNoScheduleTaint()
+	Expect(err).ToNot(HaveOccurred(), "failed to list nodes without NoSchedule taint")
+	logf.Log.Info("nodes without NoSchedule taint", "nodes", nodesWithoutTaint)
+
+	// get the no of replicas in lvm-controller deployment
+	// Scale down the lvm-controller deployment
+	// Check that lvm-controller pods has been terminated successfully
+	logf.Log.Info("Scale down and get the no of replicas in lvm-controller deployment")
+	lvmControllerName := e2e_config.LvmEngineControllerDeploymentName
+	lvmControllerOrgReplica, err = k8sinstall.ScaleLvmControllerViaHelm(0)
+	Expect(err).To(BeNil(), "failed to scale down deployment %s, error: %v", lvmControllerName, err)
+	Expect(lvmControllerOrgReplica).ShouldNot(BeZero(), "lvm controller replica count should not be zero")
+	logf.Log.Info("Lvm controller deployment", "name", lvmControllerName, "original replica", lvmControllerOrgReplica)
+
+	// create sc and pvc
+	logf.Log.Info("create sc, pvc, fio pod with snapshot as source")
+	err = app.CreateVolume()
+	Expect(err).To(BeNil(), "failed to create pvc")
+
+	// verify pvc to be in pending state
+	pvcPhase, err := k8stest.GetPvcStatusPhase(app.GetPvcName(), common.NSDefault)
+	Expect(err).ToNot(HaveOccurred(), "failed to get pvc phase")
+	Expect(pvcPhase).Should(Equal(coreV1.ClaimPending), "pvc phase is not pending")
+
+	// verify pvc pre condition failed events
+	isEventPresent, err := k8stest.WaitForPvcNormalEvent(app.GetPvcName(), common.NSDefault, volumeProvisionErrorMsg)
+	Expect(err).To(BeNil())
+	Expect(isEventPresent).To(BeTrue())
+
+	// sleep for 15 seconds and again verify state and events
+	logf.Log.Info("sleep for 15 seconds")
+	time.Sleep(15 * time.Second)
+
+	// verify pvc to be in pending state
+	pvcPhase, err = k8stest.GetPvcStatusPhase(app.GetPvcName(), common.NSDefault)
+	Expect(err).ToNot(HaveOccurred(), "failed to get pvc phase")
+	Expect(pvcPhase).Should(Equal(coreV1.ClaimPending), "pvc phase is not pending")
+
+	// verify pvc pre condition failed events
+	isEventPresent, err = k8stest.WaitForPvcNormalEvent(app.GetPvcName(), common.NSDefault, volumeProvisionErrorMsg)
+	Expect(err).To(BeNil())
+	Expect(isEventPresent).To(BeTrue())
+
+	// Scale up the lvm-controller deployment replica to initial replica + 1
+	err = k8stest.RestoreDeploymentReplicas(lvmControllerName, common.NSOpenEBS(), 120, lvmControllerOrgReplica+1)
+	Expect(err).To(BeNil(), "failed to scale  deployment %s, error: %v", lvmControllerName, err)
+
+	// get the no of replicas in lvm-controller deployment
+	// Scale up the lvm-controller deployment
+	logf.Log.Info("Scale up lvm-controller deployment")
+	_, err = k8sinstall.ScaleLvmControllerViaHelm(lvmControllerOrgReplica + 1)
```

Comment:
better to have a differnet function which return original count of replicas of a deployment. and it can be a generic for any deployment.